File: LeetCode/Group-字符串/1768. 交替合并字符串 Merge Strings Alternately/answer.py
from itertools import zip_longest

# 一般解法用双指针即可
# zip_longest当多个输入iterables的长度不相等的时候，zip会按照最小的长度进行迭代处理，其余补fillvalue值
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        return ''.join(a + b for a, b in zip_longest(word1, word2, fillvalue=''))

File: LeetCode/Group-字符串/1768. 交替合并字符串 Merge Strings Alternately/answer.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        int word1length = word1.length();
        int word2length = word2.length();
        int j = 0;
        char[] result = new char[word1length + word2length];
        for (int i = 0; i < word1length || i < word2length; i++) {
            if (i < word1length)
                result[j++] = word1.charAt(i);
            if (i < word2length)
                result[j++] = word2.charAt(i);
        }
        return new String(result);
    }
}